### ISSUE TITLE
Browsing albums and 

### ISSUE THREAD

**Issue by brjhaverkamp** [theothernt/AerialViews#167]
**Describe the solution you'd like...**

Hi Neil,

This might be a bit too far out of the original goal of aerialview, but I am searching high and low for an app that can simply show a photo album from my NAS, connected to android TV over SMB.  Your great tool is now in daily use as screensaver here in our living room, so I thought I'd see if you are interested in this usecase.

My problem is mainly that none of the photoviewer and file browsers I tried(and I tried a lot), can show a set of photos sorted on date taken.
I'm looking for photo (and video) viewer where you can simply navigate to a folder and show the images in chronological (date taken) order.  Or show them as a slide show.

Is this something you could add to Aerialviews?

Kind regards,

Bert

**Comment by theothernt** [comment]
Unfortunately that is out of scope for what the screensaver does at the moment.

That said, a lot of it is a limitation of Android TV. On an Android phone, there is already a file or photo picker but it was removed for Android TV. Making it would require a lot of work, and I have a number of other things to implement first.

There is another issue. Loading a list of files (photos, videos) versus loading a list of files plus there metadata takes a lot longer. It is possible, but I would have to implement caching for local, SMB, WebDAV, etc providers.

Over the next few releases, I'm hoping to improve the photos feature - but even trying to show photos in date order will be tricky with how files are currently read... so I'll see how it goes. Also, I recently found out a user who has over 120K photos so Aerial Views takes 1-2 minutes to start. I might have to look at that too!

**Comment by theothernt** [comment]
Question for you - do you have all your photos in a set of folder? have you thought about using some kind of app (runs on a local server or online) to organise all your photos, create albums, etc?

**Comment by brjhaverkamp** [comment]
Hi Neil, what do you mean? I do indeed have my photos in a set of folders. Organized by album, each album being a folder. I like to keep everything local. So online is not an option. Call me oldfasioned, or paranoid. I like to be in control :-)
I tried plex to serve the photos from my NAS, but it has some quirks. Photo albums are not the prime focus of those media server applicatons.  Kodi does a decent job, at least it shows them all with the right orientation, also the videos. But it failed on showing them in order of "create date". The work-around I now use is to rename all photos and videos in an album with their create date. That is a simple operation in Digikam.  It shouldn't be needed, but it does the job for now.

**Comment by theothernt** [comment]
In the next couple of updates, I'll be adding support for Immich which is a free, open source, media library you can host yourself - [https://immich.app/](https://immich.app/)

As a media library, you can create albums then have Aerial Views only show certain albums.

**Comment by brjhaverkamp** [comment]
Hi Neil,

I looked at immich.app. I like the name:-) Nice spelling twist. But I miss the usecase. At least for my case. I get all images (immiches) to my server via Nextcloud auto-upload apps from all phones in the family. When on the server, I can use any application on my laptop to sort them in albums the way I want them. The storage is simply in folders and a directory hierarchy.  I'm not sure how immiche can make this easier.

The only weak link I have is showing the pictures on my Big screen. For screensavers and random images your wonderful AerialViews has solved this 100%. For showing a particular album, in the right order, I now have a workaround via renaming the photos and using Kodi. One day, some day, I hope to not need the renaming and being able to do the sorting on the create-date timestamp.

Let me know if I missed something

Regards,

Bert

**Comment by theothernt** [comment]
> I looked at immich.app. I like the name:-) Nice spelling twist. But I miss the usecase. At least for my case. I get all images (immiches) to my server via Nextcloud auto-upload apps from all phones in the family. When on the server, I can use any application on my laptop to sort them in albums the way I want them. The storage is simply in folders and a directory hierarchy. I'm not sure how immiche can make this easier.

Currently, Aerial Views can only get a flat list of files. Even the other of the files is controlled by how the server lists them. Using a proper library or management system means you can order the photos properly and group them in albums. Then, Aerial Views is able to use that extra information to order the photos, know which are landscape/portrait, etc - it's all about the metadata!

Anyway, if a flat set of files in folders works for you - that's fine - I'm just hoping to add alternatives as new features get added :)

**Comment by brjhaverkamp** [comment]
Well, connected is maybe not the right term.
I have all my photos in a folder on my NAS. That one is synced with nextcloud to my desktop and laptop computer.
So I can sort and select photos on either device and they will be synced back to the NAS automatically.
(I mainly use digikam for the sorting and editting part.)
I have made the photo folder on the NAS also an NFS share. so I can use aerialviews to show it on my TV as screensaver as well. 

For viewing the photos in the right order, I have now found a function in digikam to rename the files based on timestamp. Now the alphabetical order is identical to the timestamp order. Not ideal, because I have to remember to rerun it every time I add a picture, but that is not that often, so I can live with it.
